src/operations/map/MapOperations.ts

```diff
@@ -86,44 +107,97 @@ export class MapOperations {
     });
   }
 
-  static setMapInteractionFunctionality(baseMap: mapboxgl.Map): void {
+  static setMapInteractionFunctionality(
+    { countries, ipcData, selectedMapType }: MapProps,
+    baseMap: mapboxgl.Map
+  ): void {
     let hoveredPolygonId: string | number | undefined;
+    const popup = new mapboxgl.Popup({ closeButton: false, closeOnClick: false });
 
-    baseMap.on('mousemove', 'country-fills', (e) => {
-      if (e.features && e.features.length > 0 && (e.features[0] as unknown as CountryMapData).properties.interactive) {
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
-        }
-        hoveredPolygonId = e.features[0].id;
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
+    baseMap.on('mousemove', 'country-fills', (e: any) => {
+      const feature = e.features && (e.features[0] as CountryMapData);
+
+      if (hoveredPolygonId && (!feature || hoveredPolygonId !== feature.id)) {
+        baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
+        hoveredPolygonId = undefined;
+        baseMap.getCanvas().style.cursor = '';
+      }
```

Comment:
When you hover a country the country color is set to nothing, this needs to be fixed, maybe that is causing the issue

Reply:
what issue exactly?